influxql/parser.go

```diff
@@ -493,6 +503,63 @@ func (p *Parser) parseSelectStatement() (*SelectStatement, error) {
 	return stmt, nil
 }
 
+// targetRequirement specifies whether or not a target clause is required.
```

Comment:
Boolean parameters are generally considered bad practice any any language.  Think about how it reads in code when you see a call to the function...
```
target, err := parseTarget(targetRequired)
```
vs
```
target, err := parseTarget(true)
```

https://www.google.com/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=boolean%20function%20parameters